tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarymindmap.code.tex

```diff
@@ -110,17 +101,41 @@
 }}%
 \tikzset{every circle connection bar/.style={}}%
 
+\pgfkeys{/pgf/decoration/start flare length/.initial=0pt}%
+\pgfkeys{/pgf/decoration/end flare length/.initial=0pt}%
+
 \def\tikz@compute@circle@radii{%
   \pgf@process{\pgfpointtransformed{\pgfpointanchor{\tikztostart}{center}}}%
   \pgf@xa=\pgf@x%
   \pgf@process{\pgfpointtransformed{\pgfpointanchor{\tikztostart}{west}}}%
   \advance\pgf@xa by-\pgf@x%
   \pgfkeys{/pgf/decoration/start radius/.expanded=\the\pgf@xa}%
+  \pgfutil@tempdima=.5\pgf@xa\relax%
   \pgf@process{\pgfpointtransformed{\pgfpointanchor{\tikztotarget}{center}}}%
   \pgf@xa=\pgf@x%
   \pgf@process{\pgfpointtransformed{\pgfpointanchor{\tikztotarget}{west}}}%
   \advance\pgf@xa by-\pgf@x%
   \pgfkeys{/pgf/decoration/end radius/.expanded=\the\pgf@xa}%
+  \advance\pgfutil@tempdima by.5\pgf@xa\relax%
+  % Now tempdima = .5 start radius + .5 end radius = nominal sum of flare lengths.
+  \pgfpointdiff{\pgfpointanchor{\tikztostart}{center}}%
+               {\pgfpointanchor{\tikztotarget}{center}}%
+  \pgfmathparse{veclen(\the\pgf@x,\the\pgf@y)}%
+  \pgfmathsetlength\pgfutil@tempdimb{\pgfmathresult}%
+  \advance\pgfutil@tempdimb by-2\pgfutil@tempdima%
+  % Now tempdimb = room available for flares.
+  \ifdim\pgfutil@tempdimb>\pgfutil@tempdima\relax%
+    \pgfutil@tempdimb=\pgfutil@tempdima\relax%
+  \fi%
+  % Now tempdimb = actual sum of flare lengths.
+  \pgf@xa=\pgfkeysvalueof{/pgf/decoration/start radius}\relax%
+  \pgf@xa=\dimexpr .5\pgf@xa * \pgfutil@tempdimb / \pgfutil@tempdima\relax%
```

Comment:
This does not do what you expect.  Try for example this:
```latex
\pgfutil@tempdima=1pt
\pgfutil@tempdimb=2pt
\the\dimexpr \pgfutil@tempdima / \pgfutil@tempdimb \relax % 0.00002pt
```

Reply:
This is probably safe.
```suggestion
  \pgfmathdivide@{\pgf@sys@tonumber\pgfutil@tempdima}{\pgf@sys@tonumber\pgfutil@tempdimb}%
  \pgf@xa=.5\pgf@xa
  \pgf@xa=\pgfmathresult\pgf@xa
```

Reply:
Oh wait, your version actually *does* work because of a corner case in `\dimexpr` where it evaluates `a * b / c` without returning the intermediate result to TeX.

Reply:
I've rewritten it using pgfmathdivide@, multiply, and add.